addons/website_livechat/models/discuss_channel.py

```diff
@@ -8,6 +8,10 @@
 class DiscussChannel(models.Model):
     _inherit = 'discuss.channel'
 
+    is_pending_chat_request = fields.Boolean(
```

Comment:
It's a bit annoying to add an extra field just for this, can we maybe check if the visitor already has a member?

Reply:
I don't think so. Before this PR, any channel opened would be considered a pending chat requests, overriding the local state with the `force_thread` mechanism on every page load. Now that CW are saved in the local storage, we need to open chat windows explicitly when a chat request is made. I want to restrict this forceful opening as much as possible. The visitor is added to the channel directly when the operator starts the chat request, and I think it's required to have proper channel information such as name, members, etc.